Dockerfile

```diff
@@ -0,0 +1,82 @@
+FROM centos:7
+# this is joshua-agent
+
+RUN yum repolist && \
+    yum install -y \
+        centos-release-scl-rh \
+        epel-release \
+        scl-utils \
+        yum-utils && \
+    yum -y install \
+        devtoolset-8 \
+        devtoolset-8-libubsan-devel \
+        devtoolset-8-valgrind-devel \
+        gettext \
+        golang \
+        java-11-openjdk-devel \
+        mono-core \
+        net-tools \
+        rh-python38 \
+        rh-python38-python-devel \
+        rh-python38-python-pip \
+        rh-ruby27 \
+        rh-ruby27-ruby-devel && \
+    source /opt/rh/devtoolset-8/enable && \
+    source /opt/rh/rh-python38/enable && \
+    source /opt/rh/rh-ruby27/enable && \
+    pip3 install \
+        python-dateutil \
+        subprocess32 \
+        psutil && \
+    gem install ffi --platform=ruby && \
+    groupadd -r joshua -g 4060 && \
+    useradd \
+        -rm \
+        -d /home/joshua \
+        -s /bin/bash \
+        -u 4060 \
+        -g joshua \
+        joshua && \
+    mkdir -p /var/joshua && \
+    chown -R joshua:joshua /var/joshua && \
+    rm -rf /tmp/*
+
+COPY childsubreaper/ /opt/joshua/install/childsubreaper
+COPY joshua/ /opt/joshua/install/joshua
+COPY setup.py /opt/joshua/install/
+
+RUN source /opt/rh/devtoolset-8/enable && \
+    source /opt/rh/rh-python38/enable && \
+    source /opt/rh/rh-ruby27/enable && \
+    pip3 install /opt/joshua/install && \
+    rm -rf /opt/joshua/install
+
+ARG OLD_FDB_BINARY_DIR=/app/deploy/global_data/oldBinaries/
+ARG OLD_TLS_LIBRARY_DIR=/app/deploy/runtime/.tls_5_1/
+ARG FDB_VERSION="6.2.29"
+RUN if [ "$(uname -p)" == "x86_64" ]; then \
+        mkdir -p ${OLD_FDB_BINARY_DIR} \
```

Comment:
Can we move this `mkdir` out of `if` statement as they're not platform dependent?
Not a big deal, but we have a wrapper Dockerfile for aarch64 which does `COPY` old binaries to the oldBinaries directory.

Reply:
the COPY directive creates the destination directory (path). We're relying on that feature with the COPYs at lines 44:46.